core/src/main/java/hudson/slaves/Cloud.java

```diff
@@ -156,13 +158,71 @@ public ACL getACL() {
      *      just needs to return {@link PlannedNode}s that each contain an object that implements {@link Future}.
      *      When the {@link Future} has completed its work, {@link Future#get} will be called to obtain the
      *      provisioned {@link Node} object.
+     * @deprecated Use {@link #provision(CloudState, int)} instead.
      */
-    public abstract Collection<PlannedNode> provision(Label label, int excessWorkload);
+    @Deprecated
+    public Collection<PlannedNode> provision(Label label, int excessWorkload) {
+        String methodName = "provision";
+        if (Util.isOverridden(Cloud.class, getClass(), methodName, CloudState.class, int.class)) {
+            return provision(new CloudState(label, 0), excessWorkload);
+        } else {
+            throw new AbstractMethodError("you must override at least one of the "
+                    + Cloud.class.getSimpleName() + "." + methodName + " methods");
+        }
```

Comment:
You can create a private method to avoid duplicating this method, kinda `callIfOverriden` passing the code to call. WDYT?

Reply:
Good idea.